File: backend/quantamind_api/routes/health.py
from __future__ import annotations

from datetime import datetime, timezone

from fastapi import APIRouter

router = APIRouter(prefix="/api/v1", tags=["health"])


@router.get("/health")
def health_check() -> dict:
    return {
        "success": True,
        "data": {
            "service": "quantamind-api",
            "status": "ok",
            "timestamp": datetime.now(timezone.utc).isoformat(),
        },
        "error": None,
    }

File: frontend/src/api/health.ts
import { getJson } from './client'

export type HealthResponse = {
  success: boolean
  data: {
    service: string
    status: string
    timestamp: string
  }
  error: null | {
    code?: string
    message?: string
  }
}

export function fetchHealth() {
  return getJson<HealthResponse>('/api/v1/health')
}
